### PR TITLE
fixed bug in attr() on forms.  no longer falls back to property

### PR DESCRIPTION
**Pull request by benmonro**
now with some tests.
https://github.com/madrobby/zepto/issues/1080

### CODE REVIEW

File: test/zepto.html [hunk 1]
```diff
@@ -1965,6 +1965,16 @@
         t.assertEqual('[boom]', el.data('invalid'))
       },
 
+      testFormAttrNull: function(t) {
+        var input = $("#attr_val");
```
Comment (mislav) [inline test/zepto.html]: We don't use semicolons

File: test/zepto.html [hunk 2]
```diff
@@ -1965,6 +1965,16 @@
...
+
+          t.assertNull(input.attr("form"), "Form should be null");
+      },
+        testFormAttrNotNull: function(t){
```
Comment (mislav) [inline test/zepto.html]: The indentation around this code is inconsistent

File: src/zepto.js [hunk 3]
```diff
@@ -286,7 +286,9 @@ var Zepto = (function() {
   }
 
   function setAttribute(node, name, value) {
-    value == null ? node.removeAttribute(name) : node.setAttribute(name, value)
+    node.setAttribute ?
```
Comment (mislav) [inline src/zepto.js]: In which case will a DOM node not have `setAttribute` defined?

File: src/zepto.js [hunk 4]
```diff
@@ -286,7 +286,9 @@ var Zepto = (function() {
...
+      (value == null ? node.removeAttribute(name) : node.setAttribute(name, value))
+      :node.prop(name, value)
```
Comment (mislav) [inline src/zepto.js]: What is the `node.prop()` method? As far as I can tell, `node` here is a plain DOM node. There is no `prop()` method in the DOM.

File: src/zepto.js [hunk 5]
```diff
@@ -12,7 +12,7 @@ var Zepto = (function() {
     tagExpanderRE = /<(?!area|br|col|embed|hr|img|input|link|meta|param)(([\w:]+)[^>]*)\/>/ig,
     rootNodeRE = /^(?:body|html)$/i,
     capitalRE = /([A-Z])/g,
-
+    booleanAttributesRE = /^(?:checked|selected|async|autofocus|autoplay|controls|defer|disabled|hidden|ismap|loop|multiple|open|readonly|required|scoped)$/i,
```
Comment (mislav) [inline src/zepto.js]: I would prefer if this was an array of strings instead of a big regular expression

File: test/zepto.html [hunk 6]
```diff
@@ -1965,6 +1965,16 @@
...
+            var input = $("#attr_val_outside_form");
+
+            t.assertEqual("attr_val_form", input.attr("form"), "Form should be null");
```
Comment (mislav) [inline test/zepto.html]: Why does this test expect that `input.attr("form")` will result in a string `"attr_val_form"`?

File: src/zepto.js [hunk 7]
```diff
@@ -622,9 +624,11 @@ var Zepto = (function() {
     },
     attr: function(name, value){
       var result
+
       return (typeof name == 'string' && !(1 in arguments)) ?
-        (!this.length || this[0].nodeType !== 1 ? undefined :
-          (!(result = this[0].getAttribute(name)) && name in this[0]) ? this[0][name] : result
+        (!this.length || this[0].nodeType !== 1 ? undefined :(
+          booleanAttributesRE.test(name) ? this[0][name] :
```
Comment (mislav) [inline src/zepto.js]: Why did you choose to return a boolean value from specific properties? jQuery doesn't seem to do it:

```js
$('<input type=checkbox checked>').attr('checked')  //=> "checked"
```

**Comment by madrobby** [comment]
Closing this because there was no response to our questions in a while. Feel free to reopen with the requested information!

**Pull request closed without merging**